src/qcodes/dataset/database_extract_runs.py

```diff
@@ -168,3 +177,196 @@ def _extract_single_dataset_into_db(
     _populate_results_table(
         source_conn, target_conn, dataset.table_name, target_table_name
     )
+
+
+def export_datasets_and_create_metadata_db(
+    source_db_path: str | Path,
+    target_db_path: str | Path,
+    export_path: str | Path | None = None,
+) -> dict[int, Literal["exported", "copied_as_is", "failed"]]:
+    """
+    Export all datasets from a source database to NetCDF files and create
+    a new database file containing only metadata (no raw data) for those exported
+    datasets. Datasets that cannot be exported to NetCDF will be transferred
+    as-is to the new database file.
+
+    This function is useful for reducing the size of database files by offloading
+    raw data to NetCDF files while preserving all metadata information in a database file.
+
+    Args:
+        source_db_path: Path to the source database file
+        target_db_path: Path to the target database file. Will be created if it doesn't exist.
```

Comment:
Should we clarify what happens if the file exists

Reply:
```suggestion
        target_db_path: Path to the target database file that will be created. Error is raised if it already exist.
```

Reply:
definitely.